coreax/solvers/composite.py

```diff
@@ -183,4 +223,11 @@ def _binary_tree(_input_data: Data) -> np.ndarray:
         return node_indices.reshape(n_leaves, -1).astype(np.int32)
 
     indices = jax.pure_callback(_binary_tree, result_shape, padded_dataset)
-    return dataset[indices]
+    return dataset[indices], jnp.arange(len(dataset))[indices]
```

Comment:
I think the return value should be `padded_dataset[indices]`. This bug was in the code before this PR and is another indication that the existing tests are insufficient. Please can you modify/add a test so that this behaviour is properly tested.